mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala

```diff
@@ -527,9 +550,21 @@ private[ml] class FeedForwardModel private(
 
   override def predict(data: Vector): Vector = {
     val size = data.size
-    val result = forward(new BDM[Double](size, 1, data.toArray))
+    val result = forward(new BDM[Double](size, 1, data.toArray), true)
     Vectors.dense(result.last.toArray)
   }
+
+  override def predictRaw(data: Vector): Vector = {
+    val size = data.size
```

Comment:
This temp val of "size" is only used once, so I recommend removing it to make the code clearer.